### PR TITLE
[stable/8.6] feat: migration of spring sdk test modules

### PR DESCRIPTION
**Pull request by megglos**
## Description

Introduces spring sdk test modules from former community sdk, I integrated the latest state of these modules from
https://github.com/camunda-community-hub/spring-zeebe/tree/main/test .

**Don’t get confused by the size**, actual changes are mostly maven related to integrate the test modules into the project + renaming + adding license headers. **Only the [first](https://github.com/camunda/zeebe-process-test/pull/1284/commits/e7aaf6435d85929967f30254cde87d016dafdd7b) and [last](https://github.com/camunda/zeebe-process-test/pull/1284/commits/99487ceb913ea2b6fef3295bcea82fd181282d0c) commit are newly written and thus worth reviewing**, all others are just from the community repo history.
First commits bumps and aligns modules versions to 8.6.7-SNAPSHOT, last integrates the spring test modules.

Applied Apache 2.0 license to all files, as it was previously published under Apache 2.0 in the community project.

## Related issues

closes [#26424](https://github.com/camunda/camunda/issues/26424)

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to backport the fix

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually

Documentation:
* [ ] Javadoc has been written
* [ ] The documentation is updated

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -91,7 +92,7 @@
     <dependency.snakeyaml.version>2.2</dependency.snakeyaml.version>
     <dependency.spring.version>6.1.8</dependency.spring.version>
     <dependency.testcontainers.version>1.19.8</dependency.testcontainers.version>
-    <dependency.zeebe.version>8.6.6</dependency.zeebe.version>
+    <dependency.zeebe.version>8.6.7-SNAPSHOT</dependency.zeebe.version>
```
Comment (megglos) [inline pom.xml:95]: temporarily as is depends on this fix https://github.com/camunda/camunda/pull/26419

File: spring-test/testcontainer/pom.xml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>io.camunda</groupId>
+    <artifactId>spring-boot-starter-camunda-test-pom</artifactId>
+    <version>8.6.7-SNAPSHOT</version>
+    <relativePath>../pom.xml</relativePath>
+  </parent>
+
+  <artifactId>spring-boot-starter-camunda-test-testcontainer</artifactId>
```
Comment (megglos) [inline spring-test/testcontainer/pom.xml:11]: A bit unfortunate naming inconsistency with `qa/testcontainers`, missing the plural `s`. However for better pre 8.6 migration experience I would keep this artifacts name singular. Wdyt?

**Comment by nicpuppa** [comment]
Does it make sense to bring all the community repo commits ? @megglos

File: pom.xml [hunk 3]
```diff
@@ -91,7 +92,7 @@
     <dependency.snakeyaml.version>2.2</dependency.snakeyaml.version>
     <dependency.spring.version>6.1.8</dependency.spring.version>
     <dependency.testcontainers.version>1.19.8</dependency.testcontainers.version>
-    <dependency.zeebe.version>8.6.6</dependency.zeebe.version>
+    <dependency.zeebe.version>8.6.7-rc1</dependency.zeebe.version>
```
Comment (nicpuppa) [inline pom.xml:95]: ❓ Why this version ?

Comment (megglos) [reply pom.xml:95]: I needed a release with https://github.com/camunda/camunda/commit/7c9485a27be1dfe3636a572465efedaaa743ef45 will be replaced with the official 8.6.7 patch once available next week.

**Comment by megglos** [comment]
> Does it make sense to bring all the community repo commits ? @megglos

I think so, we have the whole history if we wanna check why and when something was done + acknowledge the contributions  with keeping the history.

**Review by nicpuppa — APPROVED** [review]

**Pull request merged**